src/client/components/pages/Courses/ViewModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {
+  FunctionComponent,
+  ReactElement,
+  useEffect,
+  useRef,
+} from 'react';
+import {
+  Modal,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Button,
+  VARIANT,
+} from 'mark-one';
+
+interface ViewModalProps {
+  /**
+   * True if the modal should be open
+   */
+  isVisible: boolean;
+
+  /**
+   * Handler to be invoked when the modal closes
+   */
+  onClose: () => void;
+}
+
+/**
+* Displays the [[SemesterTable]] used for customizing the currently visible
+* columns in the [[CourseInstanceTable]]
+*/
+const ViewModal: FunctionComponent<ViewModalProps> = ({
+  isVisible,
+  onClose,
+  children,
+}): ReactElement => {
+  /**
+   * Ref to attach to the internal modal header
+   */
+  const modalHeaderRef = useRef<HTMLHeadingElement>(null);
+
+  useEffect((): void => {
+    if (isVisible) {
+      modalHeaderRef.current.focus();
+    }
+  }, [isVisible]);
+
+  return (
+    <Modal
+      ariaLabelledBy="view-modal-header"
+      isVisible={isVisible}
+      closeHandler={() => {
+        onClose();
+      }}
```

Comment:
@rmainseas Just to confirm, this would allow the selections to be saved without clicking the "Done" button. Is this the intended behavior? I thought that the "done" button was acting as a "save" button. However, it seems like we intend the columns selected to change in the background as you select them in the modal, so it seems like this current implementation is ok (just wanted to confirm with you though).

Reply:
Yep. That's intentional - I believe that was the decision we came to since having the changes only commit when you press "Done" would require maintaining two pieces of separate state (and thus additional complexity). If someone really gets themselves into a mess and somehow hides everything or whatever, the changes aren't persisted outside of react state - so they can just reload the page and "un-stick" themselves (so-to-speak).

Reply:
Ah, I just realized what you mean about this not having the unsaved changes behavior....

I'm not sure if we talked about that aspect of things or not....personally, I'd say it was probably fine since clicking "Done" simply dismisses the modal anyway and besides - there aren't really any unsaved changes since the changes the user makes take effect immediately.

Reply:
Ok, it seems like this makes the most sense for now, especially if "done" is just dismissing the modal.